plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/parser/DockerImageIdentifierParser.java

```diff
@@ -26,59 +26,74 @@
+    private static final String  REPOSITORY         = NAME_COMPONENT + "(?:/" + NAME_COMPONENT + ")*";
+    private static final String  TAG                = "[\\w][\\w.-]*";
+    private static final String  DIGEST             = "[\\w+.:-]+";
+    private static final String  NAME               = "(?:" + REGISTRY + "/)?" + REPOSITORY;
+    private static final String  REFERENCE          = NAME + "(?::" + TAG + ")?" + "(?:@" + DIGEST + ")?";
+    private static final Pattern IMAGE_PATTERN      = Pattern.compile(REFERENCE);
 
+    /**
+     * Validates and parse docker image reference into object that holds reference components
+     *
+     * @param image image reference to parse
+     * @throws DockerFileException
+     * @throws IllegalArgumentException
+     */
     public static DockerImageIdentifier parse(final String image) throws DockerFileException {
-        if (image == null) {
-            throw new IllegalArgumentException("Null argument value is forbidden");
+        if (image == null || image.isEmpty()) {
+            throw new DockerFileException("Null and empty argument value is forbidden");
         }
 
+        Matcher matcher = IMAGE_PATTERN.matcher(image);
+        if (!matcher.matches()) {
+            throw new DockerFileException("Provided image reference is invalid");
+        }
+
+        DockerImageIdentifier.DockerImageIdentifierBuilder identifierBuilder = DockerImageIdentifier.builder();
         String workingCopyOfImage = image;
-        String digest = "";
-        String tag = "";
 
-        // find digest
+        // extract digest
         int index = workingCopyOfImage.lastIndexOf('@');
         if (index != -1) {
-            digest = workingCopyOfImage.substring(index + 1);
+            String digest = workingCopyOfImage.substring(index + 1);
             if (!digest.isEmpty()) {
                 workingCopyOfImage = workingCopyOfImage.substring(0, index);
+                identifierBuilder.setDigest(digest);
             }
         }
 
-        // find tag
+        // extract tag
         index = workingCopyOfImage.lastIndexOf(':');
         if (index != -1) {
             if (workingCopyOfImage.lastIndexOf('/') < index) {
-                tag = workingCopyOfImage.substring(index + 1);
+                String tag = workingCopyOfImage.substring(index + 1);
                 if (!tag.isEmpty()) {
                     workingCopyOfImage = workingCopyOfImage.substring(0, index);
+                    identifierBuilder.setTag(tag);
                 }
             }
         }
 
-        Matcher matcher = IMAGE_PATTERN.matcher(workingCopyOfImage);
-        if (!matcher.matches()) {
-            throw new DockerFileException("Provided image reference is invalid");
+        index = workingCopyOfImage.indexOf('/');
+        String beforeSlash = index > -1 ? workingCopyOfImage.substring(0, index) : "";
+        if (beforeSlash.isEmpty() || (!beforeSlash.contains(".") &&
```

Comment:
It is not clear from the context what this `contains(...)` invocations actually check, consider extracting this condition into the method or adding a comment explaining the context of the condition.